services/httppost/httpposttest/server.go

```diff
@@ -0,0 +1,53 @@
+package httpposttest
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/influxdata/kapacitor/alert"
+)
+
+type Server struct {
+	ts     *httptest.Server
+	URL    string
+	data   []Request
+	closed bool
+}
+
+func NewServer(headers map[string]string) *Server {
+	s := new(Server)
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		req := Request{MatchingHeaders: true}
+		for k, v := range headers {
+			nv := r.Header.Get(k)
+			if nv != v {
+				req.MatchingHeaders = false
+			}
+		}
+		req.Data = alert.Data{}
+		dec := json.NewDecoder(r.Body)
+		dec.Decode(&req.Data)
+		s.data = append(s.data, req)
+	}))
+	s.ts = ts
+	s.URL = ts.URL
+	return s
+}
+
+type Request struct {
+	MatchingHeaders bool
+	Data            alert.Data
```

Comment:
This seems incorrect now. The posted data could be either alert data or just a raw row of data. Should we try and capture both or  alternately we could just capture the entire body and let the consumer of this test type decode it.